backend/src/utils/userUtils.ts

```diff
@@ -53,7 +53,10 @@ export const getUserName = async (
     const userOauth = await getUser(request, customObjectsApi);
     return userOauth.metadata.name;
   } catch (e) {
-    const userCluster = (currentUser.username || currentUser.name)?.split('/')[0];
-    return !userCluster || userCluster === 'inClusterUser' ? DEFAULT_USERNAME : userCluster;
+    if (DEV_MODE) {
+      return (currentUser.username || currentUser.name)?.split('/')[0];
+    }
+    fastify.log.error(`Error retrieving username, ${e.toString()}`);
```

Comment:
Avoid `e.toString()` -- we should look to do `e.response.body.message` to get the k8s error. If it's our own error throw from a lower method, `e.message`... support both instead of `e.toString()` as it will most likely swallow the actual error and not print it in the logs.